build/gulp/context.js

```diff
@@ -1,9 +1,9 @@
-var fs = require('fs');
 var argv = require('yargs')
     .default('uglify', false)
     .argv;
 
-var productVersion = readVersion(),
+var productVersion = require('../../package.json').version,
+    scriptVersion = productVersion,
```

Comment:
Let's get rid of `scriptVersion`  variable at all.

Reply:
We need unchanged (scriptVersion) version from product.json, but productVersion will be changed.

Reply:
Do we need all three?
`productVersion`, `scriptVersion`, `packageVersion`

Reply:
Taking into account this:
https://github.com/DevExpress/DevExtreme/blob/941bc99c0fccfbb9ca9720e54d880c62f841cb62/build/gulp/context.js#L46-L48

looks like we can kill `scriptVersion` and  use `packageVersion` instead now.

Reply:
> 
> 
> Taking into account this:
> 
> [DevExtreme/build/gulp/context.js](https://github.com/DevExpress/DevExtreme/blob/941bc99c0fccfbb9ca9720e54d880c62f841cb62/build/gulp/context.js#L46-L48)
> 
> Lines 46 to 48 in [941bc99](/DevExpress/DevExtreme/commit/941bc99c0fccfbb9ca9720e54d880c62f841cb62)
>  if(packageVersion !== scriptVersion) { 
>      throw 'Product version mismatch in ' + SCRIPT_VERSION_FILE; 
>  } 
> 
> looks like we can kill `scriptVersion` and use `packageVersion` instead now.

Your example is just reading from package.json and validation. Look at https://github.com/DevExpress/DevExtreme/blob/941bc99c0fccfbb9ca9720e54d880c62f841cb62/build/gulp/context.js#L23 and https://github.com/DevExpress/DevExtreme/blob/941bc99c0fccfbb9ca9720e54d880c62f841cb62/build/gulp/context.js#L24.

Hotfix, for example, will have:

package version: `18.2.4-pre-10000`
product version: `18.2.3 (build 10000)`
script version: `18.2.3` <--DevExpress.VERSION